lib/corpus.py

```diff
@@ -300,6 +299,18 @@ def _dump_internal(self, out: TextIO, indentation: str = ""):
         else:
             out.write("%s</subcorpus>\n" % (indentation,))
 
+    def get_segment_mapping(self) -> Dict[str, Segment]:
+        """
+        :return: Mapping from segment fullnames to actual segments.
+        """
+        return {seg.fullname(): seg for seg in self.segments()}
+
+    def get_recording_mapping(self) -> Dict[str, Recording]:
+        """
+        :return: Mapping from recording fullnames to actual recordings.
+        """
+        return {rec.fullname(): rec for rec in self.recordings}
```

Comment:
```suggestion
        return {rec.fullname(): rec for rec in self.all_recordings()}
```
in case of (nested) sub-corpora the current implementation will only contain the top-level recordings while my proposed change would also descend into sub-corpora and get all recordings.